From 982e181bf6c06a1ea40bc799cada20a2bc04b354 Mon Sep 17 00:00:00 2001
From: Pawit Pornkitprasan <p.pawit@gmail.com>
Date: Wed, 10 Dec 2014 17:35:17 +0700
Subject: [PATCH] sepolicy: allow system apps to access ASEC

Required for Settings to show name/icon of apps on sd card

(permission copied from untrusted_app)

Change-Id: Ib2b3bee4dfb54ad5e45b392fd9bfd65add4a00bf
---
 sepolicy/sepolicy.mk   | 1 +
 sepolicy/system_app.te | 3 +++
 2 files changed, 4 insertions(+)
 create mode 100644 sepolicy/system_app.te

diff --git a/sepolicy/sepolicy.mk b/sepolicy/sepolicy.mk
index 1a9b343..d916a04 100644
--- a/sepolicy/sepolicy.mk
+++ b/sepolicy/sepolicy.mk
@@ -25,6 +25,7 @@ BOARD_SEPOLICY_UNION += \
     su.te \
     sysinit.te \
     system.te \
+    system_app.te \
     ueventd.te \
     vold.te \
     mac_permissions.xml
diff --git a/sepolicy/system_app.te b/sepolicy/system_app.te
new file mode 100644
index 0000000..05907f1
--- /dev/null
+++ b/sepolicy/system_app.te
@@ -0,0 +1,3 @@
+# ASEC
+allow system_app asec_apk_file:dir { getattr };
+allow system_app asec_apk_file:file r_file_perms;
-- 
1.9.3 (Apple Git-50)

